src/subscript-loc.c

```diff
@@ -158,10 +158,11 @@ r_obj* int_as_location(r_obj* subscript,
         case LOC_ZERO_IGNORE: break;
         }
       } else if (abs(elt) > n) {
-        if (opts->loc_oob == LOC_OOB_ERROR) {
-          stop_subscript_oob_location(subscript, n, opts);
+        switch (opts->loc_oob) {
+        case LOC_OOB_ERROR: stop_subscript_oob_location(subscript, n, opts);
+        case LOC_OOB_EXTEND:
+        case LOC_OOB_REMOVE: ++n_oob; break;
         }
-        ++n_extend;
```

Comment:
I strongly prefer the switch statement in these kinds of paths. If we add a new `LOC_OOB_*` option then the compiler will throw a warning that we haven't handled that case, so we can't forget about it